gax-grpc/src/main/java/com/google/api/gax/grpc/OperationCallable.java

```diff
@@ -27,148 +27,112 @@
  * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
-package com.google.api.gax.grpc;
+package com.google.api.gax.rpc;
 
-import static com.google.common.base.Preconditions.checkNotNull;
-
-import com.google.api.core.ApiFunction;
 import com.google.api.core.ApiFuture;
 import com.google.api.core.BetaApi;
-import com.google.api.gax.retrying.RetryAlgorithm;
-import com.google.api.gax.retrying.RetryingExecutor;
-import com.google.api.gax.retrying.RetryingFuture;
-import com.google.api.gax.retrying.ScheduledRetryingExecutor;
-import com.google.longrunning.CancelOperationRequest;
-import com.google.longrunning.GetOperationRequest;
-import com.google.longrunning.Operation;
-import com.google.longrunning.OperationsClient;
-import com.google.protobuf.Any;
-import com.google.protobuf.Empty;
-import com.google.protobuf.InvalidProtocolBufferException;
-import com.google.protobuf.Message;
-import io.grpc.Status;
+import com.google.common.base.Preconditions;
+import javax.annotation.Nullable;
 
 /**
  * An OperationCallable is an immutable object which is capable of initiating RPC calls to
- * long-running API methods and returning an OperationFuture to manage the polling of the Operation
- * and getting the response.
+ * long-running API methods and returning an {@link OperationFuture} to manage the polling of the
+ * Operation and getting the response.
+ *
+ * <p>It is considered advanced usage for a user to create an OperationCallable themselves. This
+ * class is intended to be created by a generated client class, and configured by instances of
+ * OperationCallSettings.Builder which are exposed through the client settings class.
  */
 @BetaApi
-public final class OperationCallable<
-    RequestT, ResponseT extends Message, MetadataT extends Message> {
-  private final UnaryCallable<RequestT, Operation> initialCallable;
-  private final ClientContext clientContext;
-  private final RetryingExecutor<Operation> executor;
-  private final OperationsClient operationsClient;
-  private final ApiFunction<Operation, ResponseT> responseTransformer;
-  private final ApiFunction<Operation, MetadataT> metadataTransformer;
+public final class OperationCallable<RequestT, ResponseT, MetadataT, OperationT>
+    implements OperationCallableImpl<RequestT, ResponseT, MetadataT, OperationT> {
+  private final OperationCallableImpl<RequestT, ResponseT, MetadataT, OperationT> callable;
```

Comment:
As we discussed offline, please change the "Impl" for the interfaces (for example by introducing the abstract class for the callables)